src/popup.tsx

```diff
@@ -0,0 +1,64 @@
+import { useStorage } from "@plasmohq/storage/hook";
+import { useCallback, useState } from "react"
+import { storage_key } from "~storage";
+import React from "react";
+
+function validate(address: string) {
+  if (!/^https?:\/\/[^/]+$/.test(address)) {
+    return "Invalid address. Please provide a valid URL pointing to a Gitpod installation. It must not have any pathes or trailing slashes."
+  }
+  return undefined;
+}
+
+function IndexPopup() {
+  const [address, setAddress] = useStorage<string>(storage_key, "https://gitpod.io");
+  const [error, setError] = useState<string>();
+  const updateAddress = useCallback((address: string) => {
+    address = address.trim();
+    address = address.endsWith("/") ? address.slice(0, -1) : address;
+    const err = validate(address);
+    setError(err);
+    setAddress(address);
+  }, [setAddress, setError]);
+
```

Comment:
in a follow up, I'll try to add the [Desktop App parser](https://github.com/gitpod-io/gitpod-desktop-app/blob/main/src/utils/parseEndpoint.ts) here, so more formats are accepted.

Reply:
https://linear.app/gitpod/issue/EXP-625/improve-setting-the-gitpod-url